packages/live-share-inking/src/core/InkingManager.ts

```diff
@@ -0,0 +1,1260 @@
+        this._isUpdating = true;
+    }
+
+    /**
+     * Unblocks the emission of update events and flushes the change log, leading to a single
+     * `StrokesAddedEvent` and/or a single `StrokesRemovedEvent`.
+     */
+    public endUpdate() {
+        if (this._isUpdating) {
+            this._isUpdating = false;
+
+            this.flushChangeLog();
+        }
+    }
+
+    /**
+     * Starts listening to pointer input.
+     */
+    public activate(): void {
+        this._inputProvider.activate();
+    }
+
+    /**
+     * Stops listening to pointer input.
+     */
+    public deactivate(): void {
+        this._inputProvider.deactivate();
+    }
+
+    /**
+     * Clears the canvas.
+     */
+    public clear() {
+        this._strokes.clear();
+
+        this.scheduleReRender();
+
+        this.notifyClear();
+    }
+
+    /**
+     * Starts a new wet stroke which will be drawn progressively on the canvas. Multiple wet strokes
+     * can be created at the same time and will not interfere with each other.
+     * @param strokeType The type of the stroke to start.
+     * @param startPoint The starting point of the stroke.
+     * @param options Creation options, such as is, points, brush...
+     * @returns An IWetStroke object representing the ongoing stroke.
+     */
+    public beginWetStroke(strokeType: StrokeType, startPoint: IPointerPoint, options?: IStrokeCreationOptions): IWetStroke {
+        const canvas = strokeType === StrokeType.laserPointer ? new LaserPointerCanvas(this._canvasPoolHost) : new WetCanvas(this._canvasPoolHost);
+        canvas.resize(this.clientWidth, this.clientHeight);
+        canvas.offset = this.offset;
+        canvas.scale = this.scale;
+
+        const stroke = new InkingManager.WetStroke(
+            this,
+            canvas,
+            strokeType,
+            options);
+
+        stroke.addPoints(startPoint);
+
+        return stroke;
+    }
+
+    /**
+     * Retrieves an existing stroke from the drawing.
+     * @param id The id of the stroke to retrieve.
+     * @returns The stroke with the specified id.
+     */
+    public getStroke(id: string): IStroke | undefined {
+        return this._strokes.get(id);
+    }
+
+    /**
+     * Adds a stroke to the drawing.
+     * @param stroke The stroke to add. If a stroke with the same id already exists
+     * in the drawing, is it replaced.
+     * @param options Options allowing the caller to force a re-render and/or bock the
+     * emission of `StrokesAddedEvent`.
+     */
+    public addStroke(stroke: IStroke, options?: IAddRemoveStrokeOptions) {
+        this.internalAddStroke(stroke, options);
+
+        if (!this._isUpdating) {
+            this.flushChangeLog();
+        }
+    }
+
+    /**
+     * Removes a stroke from the drawing.
+     * @param id The id of the stroke to remove. If the stroke doesn't exist, nothing
+     * happens.
+     * @param options Options allowing the caller to force a re-render and/or bock the
```

Comment:
```suggestion
     * @param options Options allowing the caller to force a re-render and/or block the
```

Reply:
The l on my keyboard doesn't work properly, I swear.